### PR TITLE
feat:[PLG-312] Enable Red Hat ACS On Kubernetes Cluster Policy

### PR DESCRIPTION
**Pull request by akashpaladin**
# Description

feat:[PLG-312] Enable Red Hat ACS On Kubernetes Cluster Policy
[PLG-312]
Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Chore (no code changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

image

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] My commit message/PR follows the contribution guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# **Other information**:

List any documentation updates that are needed for the Wiki

[PLG-312]: https://paladincloud.atlassian.net/browse/PLG-312?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PLG-312]: https://paladincloud.atlassian.net/browse/PLG-312?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by plyubich — APPROVED** [review]

File: jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/redhatpolicies/misc/RedhatNotScanningClusterPolicy.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            logger.error("Unable to scan cluster", e);
+            throw new RuleExecutionFailedExeption(e.getMessage());
+        }
+        logger.debug("========RedhatNotScanningClusterPolicy ended=========");
+        return new PolicyResult(PacmanSdkConstants.STATUS_SUCCESS, PacmanRuleConstants.SUCCESS_MESSAGE);
+    }
+
+    @Override
+    public String getHelpText() {
+        return "Checks for the presence of vulnerabilities.";
```
Comment (santhosh-challa) [inline jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/redhatpolicies/misc/RedhatNotScanningClusterPolicy.java:98]: Are we using this anywhere? if yes, we may need to update this. @akashpaladin

Comment (akashpaladin) [reply jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/redhatpolicies/misc/RedhatNotScanningClusterPolicy.java]: hi @santhosh-challa I have updated the help text, and I haven't seen this method used

**Review by santhosh-challa — DISMISSED** [review]
LGTM

File: jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/redhatpolicies/misc/RedhatNotScanningClusterPolicy.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                List<LinkedHashMap<String, Object>> issueList = new ArrayList<>();
+                LinkedHashMap<String, Object> issue = new LinkedHashMap<>();
+                Annotation annotation = Annotation.buildAnnotation(policyParam, Annotation.Type.ISSUE);
+                annotation.put(PacmanSdkConstants.DESCRIPTION, "Cluster - " + clusterName + " not scanned by redhat acs found");
+                annotation.put(PacmanRuleConstants.SEVERITY, severity);
+                annotation.put(PacmanRuleConstants.CATEGORY, category);
+                issue.put(PacmanRuleConstants.VIOLATION_REASON, "Cluster - " + clusterName + " not scanned by redhat acs found");
+                issueList.add(issue);
+                annotation.put("issueDetails", issueList.toString());
+                logger.debug("========RedhatNotScanningClusterPolicy ended with annotation {} : =========", annotation);
```
Comment (mkinchin-paladin) [inline jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/redhatpolicies/misc/RedhatNotScanningClusterPolicy.java:85]: do not use repetitive symbols in logs. keep logs clean, short and simple

**Review by mkinchin-paladin — APPROVED** [review]
LGTM with minor comment

File: jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/redhatpolicies/misc/RedhatNotScanningClusterPolicy.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
+/*******************************************************************************
+ *  Copyright 2023 Paladin Cloud, Inc. or its affiliates. All Rights Reserved.
```
Comment (mkinchin-paladin) [inline jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/redhatpolicies/misc/RedhatNotScanningClusterPolicy.java:2]: oh, it's already 2024 :)

**Review by mkinchin-paladin — CHANGES_REQUESTED** [review]
Requested two minor changes. After those are done - consider this PR approved.

File: jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/redhatpolicies/misc/RedhatNotScanningClusterPolicy.java (hunk 2 above)
Comment (akashpaladin) [reply jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/redhatpolicies/misc/RedhatNotScanningClusterPolicy.java]: Sure, removed

File: jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/redhatpolicies/misc/RedhatNotScanningClusterPolicy.java (hunk 3 above)
Comment (akashpaladin) [reply jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/redhatpolicies/misc/RedhatNotScanningClusterPolicy.java]: changed

**Review by arunpaladin — APPROVED** [review]

**Review by plyubich — APPROVED** [review]

File: jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/redhatpolicies/misc/RedhatNotScanningClusterPolicy.java (hunk 3 above)
Comment (mkinchin-paladin) [reply jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/redhatpolicies/misc/RedhatNotScanningClusterPolicy.java]: @akashpaladin you fix one and then added another one with 2023...

**Review by mkinchin-paladin — APPROVED** [review]
Minor comment, but rest is fine.

**Pull request merged**